specifyweb/specify/api.py

```diff
@@ -345,8 +345,24 @@ def cleanData(model, data: Dict[str, Any], agent) -> Dict[str, Any]:
             logger.warn('field "%s" does not exist in %s', field_name, model)
         else:
             cleaned[field_name] = data[field_name]
+
```

Comment:
@melton-jason 

I shifted the responsibility for standardizing the date precision to the back-end - that is where it belongs in the first place, and it is simpler (on the front-end, if resource has precision but no date, can't unset precision right away because maybe user was going to enter the date later...)

question is - is this the best place to have this code?
i.e I could see one case where it would fail (not sure if we care about that case) - if the body of the API request includes the date field, but not the precision field, than the `if field_name.endswith('precision'):` would never run

Reply:
I would agree that standardization of data should probably occur on the backend (the backend needs a little more love anyways). 

This is a fine spot for the code, I think. 
The only other place I might consider is https://github.com/specify/specify7/blob/22bda472777c22c28a660890cd84f7991169f2e4/specifyweb/specify/api.py#L408-L420
This already handles converting the SQL timestamp objects to strings (by way of https://github.com/specify/specify7/blob/22bda472777c22c28a660890cd84f7991169f2e4/specifyweb/specify/api.py#L670-L674)

However, `cleanData` also works, and is probably more intuitive at the moment. 
Would it be worth adding an assert or at the very least a console.warn to handle the case? It would (hopefully) only impact those directly using the API.